controllers/multicluster/clusterregistry/registry.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package clusterregistry
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"go.uber.org/fx"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	controllermetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/chaos-mesh/chaos-mesh/cmd/chaos-controller-manager/provider"
+	"github.com/chaos-mesh/chaos-mesh/controllers/config"
+	"github.com/chaos-mesh/chaos-mesh/controllers/multicluster/remotepodreconciler"
+	"github.com/chaos-mesh/chaos-mesh/pkg/log"
+)
+
+type remoteCluster struct {
+	app *fx.App
+
+	client.Client
+}
+
+// RemoteClusterRegistry will manage all controllers running on a remote
+// cluster. The construction of these controllers (managers) will be managed by
+// `fx`. The main process of constructing a controller manage is nearly the same
+// with the main one. The only difference is that we'll need to provide a new
+// `RestConfig`, and `Populate` the client to allow others to use its client.
+type RemoteClusterRegistry struct {
+	clusters map[string]*remoteCluster
+	logger   logr.Logger
+	client   client.Client
+
+	*sync.Mutex
```

Comment:
```suggestion
	lock *sync.Mutex
```

Using embed Mutex would expose methods `lock()` and `unlock()` to the user/developer.
It seems user/developer do not require to consider the thread-safe through the provided API, and we should hide it.